test/thrift/generator/service_test.exs

```diff
@@ -370,8 +370,6 @@ defmodule Thrift.Generator.ServiceTest do
 
   thrift_test "clients exit if they try to use a closed client", ctx do
     Process.flag(:trap_exit, true)
-
-    ref = Process.monitor(ctx.server)
```

Comment:
I think should wait for the `:DOWN` of the server before doing the client call. Otherwise the server may not have shut down before the client request. This may solve the occasional failure: https://travis-ci.org/pinterest/elixir-thrift/jobs/219331581#L484. However `ranch` will asynchronously close the listen socket/client sockets so it may not stop this error completely - the `gen_tcp` sockets are closed asynchronously when their owning process exits and `ranch` only waits for the owning processes to exit.

Note that `Process.exit(pid, :normal)` is a bit of an antipattern because it will only stop a child process that is trapping exits and adheres to OTP principles. Rather should use `Mod.stop` or if not provided`:sys.terminate` for OTP processes. `Process.exit(pid, :normal)` is unlikely to work on any non-OTP process. Therefore the only real use case for this is to stop a child Supervisor asynchronously in a test - as we are doing here but we actually want synchronous stop.